### DISCUSSION TITLE
Feature request: write to DICOM format (or at least give better errors)

### DISCUSSION THREAD

**Discussion by jsmcx** [ImageMagick/ImageMagick#6183] [Development]
### ImageMagick version

7.1.1-4

### Operating system, version and so on

Mac OS 12.6.3 

### Description

ImageMagick converts DICOM images to other formats. This function works really well. However, I have tried to convert _from_ other formats to DICOM. I use the command

magick image.jpg image.dcm

This creates a garbled file that cannot be read by any of the standard DICOM compatible apps (OsiriX, Horos, Weasis, etc). 
I have read the documentation, and I understand that DICOM is marked as a read-only format

I have 2 feature requests:
1. In the short term, return an error when trying to write to an unsupported format
2. In the longer term, please support writing to DICOM.

**Comment by urban-warrior** [comment]
ImageMagick does not write the DCM image format.  Notice how `magick -list format| grep DCM`:

```
 DCM* DCM       r--   Digital Imaging and Communications in Medicine image
```

Notice the mode of 'r--'.  That means read only.  Since there is no DCM writer, the image reverts to writing the image in the source image file format.  In your example, JPEG.

**Reply by jsmcx** [reply]
_> I have read the documentation, and I understand that DICOM is marked as a read-only format_

Writing a JPEG and giving it a .dcm extension is broken behaviour.

**Reply by urban-warrior** [reply]
We added a patch to fix this issue yesterday to the main repo.

**Comment by snibgo** [comment]
In these cases, I think it would be helpful if IM issued a warning that writing to the requested output format is not supported.

**Comment by urban-warrior** [comment]
We agree if its an explicit image format such as `dcm:image.dcm`.  We'll add a patch.  However, for implicit formats such as `image.dcm`, we will leave the behavior as it is now, but we might add a warning.

**Comment by jsmcx** [comment]
Respectfully, the current behaviour is broken. The user requests a certain action to occur, that action does not occur, and there is no feedback given.

**Reply by urban-warrior** [reply]
We added a patch to fix this issue yesterday to the main repo.

**Comment by jsmcx** [comment]
Thank you. I do appreciate you taking time to respond. 

For the benefit of future people who might see this discussion: on reading more about it, I see that DICOM is more of a metadata wrapper around a range of image formats, so I fully understand why ImageMagick treats this format as read-only. 

The correct tool for converting to DICOM format would be something like DCMTK, for example.

**Comment by nagayev** [comment]
@jsmcx For converting something to DICOM you can use [img2dcm](
https://support.dcmtk.org/docs-dcmrt/img2dcm.html)
